spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/AbstractConfigProperties.java

```diff
@@ -1,120 +0,0 @@
-/*
- * Copyright 2013-2021 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      https://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.springframework.cloud.kubernetes.commons.config;
-
-import org.springframework.boot.context.properties.bind.DefaultValue;
-
-/**
- * Abstraction over configuration properties.
- *
- * @author Ioannis Canellos
- * @author Isik Erhan
- */
-public abstract class AbstractConfigProperties {
```

Comment:
1. the idea of this PR is to move `ConfigMapConfigProperties` and `SecretsConfigProperties` to records. Since records can't extends anything, this class has to go and all of its properties are inlined into the classes that extended it: `ConfigMapConfigProperties` and `SecretsConfigProperties`.

Initially I did not like this, since I thought that, if in future, we add a property, we want it to be in both configmap and secrets and a natural thing would be is to have this abstract class. But the more I thought about it, the more it made less sense. Even if we add another property in here, without properly implementing its support in respective children means simply zero. Yeah, it helps with reading the code slightly, but also moving to records and inlining them helps too. May be I am just biased here, but it sort of makes sense for me.